enhancements/insights/rapid-recommendations.md

```diff
@@ -0,0 +1,552 @@
+(it is part of the Insights Operator GitHub repository).
+The schema can evolve over time; we intend to change it only between x-stream and y-stream versions 
+(no backports) to simplify reasoning about differences between data gathered 
+by different Insights operator versions. Changes in z-stream versions
+(backports) will still be possible in cases that we will be unable to handle at
+the level of the remote configuration service.
+
+Upon successful validation, the remote configuration will be stored in an immutable configmap, 
+and used to gather data and produce an archive. 
+The Insights Operator will use the cached configuration from the configmap when it will be unable 
+to download a valid version from the remote configuration service. 
+The Insights Operator will assume empty configuration if no copy of the remote configuration 
+will be available.
+See also [Status Reporting and Monitoring](#status-reporting-and-monitoring).
+
+Container log data is identified by namespace name, pod name and container name. 
+These names will be used when storing the data in the Insights archive 
+(see [Insights Archive Structure Changes](#insights-archive-structure-changes)). 
+The next sections elaborate on how the Insights Operator will limit gathered data: 
+[Gathering Boundaries](#gathering-boundaries), [Data Redaction/Obfuscation](#data-redactionobfuscation),
+[Limits on Processed and Stored Data](#limits-on-processed-and-stored-data) section.
+
+#### Gathering Boundaries
+
+The Insights Operator will have hardcoded boundaries for data that can be requested through 
+the remote configuration. 
+It will collect container logs only from pods in
+
+* OpenShift and Kubernetes namespaces (prefixes `openshift-` ,`kube-` and the `default`)
+
+#### Data Redaction/Obfuscation
+
+The Insights Operator will apply existing container log redaction/obfuscation settings to 
+container logs gathered using the Rapid Recommendations feature:
+
+* The Insights Operator will apply no data redaction/obfuscation by default
+* Customers can enable global obfuscation of IP addresses and cluster domain name
+
+#### Limits on Processed and Stored Data
+
+The Insights Operator will have hardcoded limits on the amount of processed and stored data:
+
+* The Insights Operator will process log messages at most 6 hours old.
+* The total amount of pods for which logs can be requested is limited by the [Gathering Boundaries](#gathering-boundaries).
+* The Insights Operator already has a limit on the archive size. The current limit is 8MB (uncompressed).
+  A new limit is being discussed separately. 
+  The current proposal is 24MB (uncompressed).
+
+
+#### Status Reporting and Monitoring
+
+The Insights Operator will add two new conditions to the ClusterOperator resource:
+
+* `RemoteConfigurationUnavailable` will indicate whether the Insights Operator can access 
+  the configured remote configuration endpoint (host unreachable or HTTP errors).
```

Comment:
How does this get bubbled up to customers, e.g. if they set up firewall rules that block or garble access?  Do we not complain, because we assume that if config-retrieval fails, archive-upload will also fail?  Or do we complain via `Degraded=True` and/or an alert, so the cluster-admin can select between "explicitly disable gathers" or "fix the networking issue"?

Reply:
Good question. Let's not assume anything. :) I think the scenario will be as follows (and yes it's probably good idea to explicitly describe it in the proposal):

- operator cannot connect to the remote config endpoint -> it will use the default fallback definition hardcoded in the binary
- data gathering runs and there's an attempt to upload the archive (and from now it's the same as today)
- if the upload fails (after few retries) then `Degraded=True` and it's up to the cluster-admin to either disable it (i.e no data, no upload) or configure your network (when there's still risk that only e.g upload is allowed, so perhaps a new Insights recommendation then?)